prusti-viper/src/encoder/spec_function_encoder.rs

```diff
@@ -150,8 +155,11 @@ impl<'p, 'v: 'p, 'tcx: 'v> SpecFunctionEncoder<'p, 'v, 'tcx> {
             posts: Vec::new(),
             body: Some(func_spec.into_iter()
                                 .map(|post| SnapshotSpecPatcher::new(self.encoder).patch_spec(post))
+                                .collect::<Result<Vec<vir::Expr>, _>>()
+                                .map_err(|err| err.with_span(self.procedure.get_span()))?
+                                .into_iter()
```

Comment:
These extra calls are a bit messy. @fpoli am I missing something with the positionless errors? Is this the best way I can map them?